modules/gcp/compute.go

```diff
@@ -581,9 +583,25 @@ func NewComputeService(t *testing.T) *compute.Service {
 func NewComputeServiceE(t *testing.T) (*compute.Service, error) {
 	ctx := context.Background()
 
-	client, err := google.DefaultClient(ctx, compute.CloudPlatformScope)
+	// Retrieve the Google OAuth token using a retry loop as it can sometimes return an error.
+	// e.g: oauth2: cannot fetch token: Post https://oauth2.googleapis.com/token: net/http: TLS handshake timeout
+	// This is loosely based on https://github.com/kubernetes/kubernetes/blob/7e8de5422cb5ad76dd0c147cf4336220d282e34b/pkg/cloudprovider/providers/gce/gce.go#L831.
+	var client *http.Client
+	var err error
+	maxRetries := 6
+	timeBetweenRetries := 10 * time.Second
+	for i := 0; i < maxRetries; i++ {
```

Comment:
Add a comment about why we aren't using `retry` module:

```
// We directly implement the retry logic here instead of using the `retry` module so that we aren't relying on modifying variables in an inner function.
```